src/libraries/SqrtPriceMath.sol

```diff
@@ -35,29 +35,34 @@ library SqrtPriceMath {
     {
         // we short circuit amount == 0 because the result is otherwise not guaranteed to equal the input price
         if (amount == 0) return sqrtPX96;
-        uint256 numerator1 = uint256(liquidity) << FixedPoint96.RESOLUTION;
+        uint256 numerator1;
+        uint256 _sqrtPX96;
+        assembly {
+            numerator1 := shl(96, liquidity)
+            _sqrtPX96 := sqrtPX96
+        }
 
         if (add) {
             unchecked {
-                uint256 product = amount * sqrtPX96;
-                if (product / amount == sqrtPX96) {
+                uint256 product = amount * _sqrtPX96;
+                if (UnsafeMath.div(product, amount) == _sqrtPX96) {
```

Comment:
why is it beneficial to call `UnsafeMath.div(product, amount)` given `product / amount` was already inside an unchecked block? whats the benefit?

Reply:
> It is not possible to disable the check for division by zero or modulo by zero using the unchecked block.

https://docs.soliditylang.org/en/latest/control-structures.html#checked-or-unchecked-arithmetic